jsonrpc/dispatcher_fuzz_test.go

```diff
@@ -0,0 +1,286 @@
+package jsonrpc
+
+import (
+	"testing"
+
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func FuzzDispatcherFuncDecode(f *testing.F) {
+	srv := &mockService{msgCh: make(chan interface{}, 10)}
+
+	dispatcher := newTestDispatcher(f,
+		hclog.NewNullLogger(),
+		newMockStore(),
+		&dispatcherParams{
+			chainID:                 0,
+			priceLimit:              0,
+			jsonRPCBatchLengthLimit: 20,
+			blockRangeLimit:         1000,
+		},
+	)
+
+	require.NoError(f, dispatcher.registerService("mock", srv))
+
+	handleReq := func(typ string, msg string) interface{} {
+		_, err := dispatcher.handleReq(Request{
+			Method: "mock_" + typ,
+			Params: []byte(msg),
+		})
+		if err != nil {
+			return err
+		}
+
+		return <-srv.msgCh
+	}
+
+	addr1 := types.Address{0x1}
+
+	seeds := []struct {
+		typ string
+		msg string
+	}{
+		{
+			"block",
+			`["earliest"]`,
+		},
+		{
+			"block",
+			`["latest"]`,
+		},
+		{
+			"block",
+			`["0x1"]`,
+		},
+		{
+			"type",
+			`["` + addr1.String() + `"]`,
+		},
+		{
+			"blockPtr",
+			`["a"]`,
+		},
+		{
+			"blockPtr",
+			`["a", "latest"]`,
+		},
+		{
+			"filter",
+			`[{"fromBlock": "pending", "toBlock": "earliest"}]`,
+		},
+		{
+			"block",
+			"[\"8\"]",
+		},
+		{
+			"block",
+			"[\"009\"]",
+		},
+		{
+			"block",
+			"10",
+		},
+	}
+
+	for _, seed := range seeds {
+		f.Add(seed.typ, seed.msg)
+	}
+
+	f.Fuzz(func(t *testing.T, typ string, msg string) {
+		handleReq(typ, msg)
+	})
+}
+
+func FuzzDispatcherBatchRequest(f *testing.F) {
+	mock := &mockWsConn{
+		SetFilterIDFn: func(s string) {
+		},
+		GetFilterIDFn: func() string {
+			return ""
+		},
+		WriteMessageFn: func(i int, b []byte) error {
+			return nil
+		},
+	}
+
+	seeds := []struct {
+		batchLimit uint64
+		blockLimit uint64
+		body       string
+	}{
+		{
+			batchLimit: 10,
+			blockLimit: 1000,
+			body: `[
+				{"id":1,"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["latest", true]},
```

Comment:
These values does not make much sense. Actual `params` should be fuzzed, not `id`. This applies to all the seeds